docs/how-to-guides/editable-installs.rst

```diff
@@ -108,6 +108,74 @@ An alternative build directory can be specified using the
 :option:`build-dir` config setting.
 
 
+Data files
+----------
+
+It is relatively common to install data files needed at runtime
+alongside the package's Python code or extension modules. For a Python
+package named ``package`` this would look like this:
+
+.. TODO update the text block below to 'meson' when
+.. meson lexer is updated to fix
+.. https://github.com/pygments/pygments/issues/2918
+
+.. code-block:: text
+
+   py = import('python').find_installation()
+
+   py.install_sources(
+       '__init__.py',
+       subdir: 'package',
+   )
+
+   install_data(
+       'data.txt',
+       install_dir: py.get_install_dir() / 'package',
+   )
+
+   custom_target(
+       output: 'uuid.txt',
+       command: [py, '-m', 'uuid''],
+       capture: true,
+       install: true,
+       install_dir: py.get_install_dir() / 'package',
+   )
+
+In most circumstances, these files can be accessed deriving their
+filesystem path from the filesystem path of the Python module next to
+them via the ``__file__`` special variable. For example, within the
+package ``__init__.py``:
+
+.. code-block:: python
+
+   import pathlib
+
+   data = pathlib.Path(__file__).parent.joinpath('data.txt').read_text()
+   uuid = pathlib.Path(__file__).parent.joinpath('uuid.txt').read_text()  # WRONG!
+
+However, this does not work when modules are not loaded from a package
+installed in the Python library path in the filesystem but with a
+special module loader, as used to implement editable installs in
+``meson-python``. In the example above, the second read would fail
+when the package is installed in editable mode.  For this reason, data
+files need to be accessed using :mod:`importlib.resources`. The code
+above should be replaced with:
+
+.. code-block:: python
+
+   import importlib.resources
+
+   data = importlib.resources.files().joinpath('data.txt').read_text()
+   uuid = importlib.resources.files().joinpath('uuid.txt').read_text()
```

Comment:
Using `files()` without an argument actually doesn't work in Python <3.12 (see [change/deprecation note [here](https://docs.python.org/3.14/library/importlib.resources.html#importlib.resources.files)). It needs a package reference as the argument - that, or use the functional `importlib.resources.read_text`.

Reply:
The functional `importlib.resources.read_text()` does not accept an empty anchor either. I think the easiest way is to depend on `importlib_resources` on older Pythons.

Reply:
If there is a way to make `importlib.resources` work for this in Python 3.9, I don't see it. If having the example depend on the `importlib-resources` backport is not an option, the only thing to do is to note that the example does not work for old Python versions and suggest the backport as a solution. I've done this in the last iteration.

Reply:
A code comment stating that seems fine to me - Python 3.9 is end-of-life already and most packages have dropped support already or will do so soon.

Reply:
I added a `.. warning`. Do you prefer that to be just a code comment?